src/gc/unix/cgroup.cpp

```diff
@@ -289,28 +427,43 @@ class CGroup
                     goto done;
                 maxLineLen = lineLen;
             }
-                   
-            // See man page of proc to get format for /proc/self/cgroup file
-            int sscanfRet = sscanf(line, 
-                                   "%*[^:]:%[^:]:%s",
-                                   subsystem_list,
-                                   cgroup_path);
-            if (sscanfRet != 2)
+
+            if (is_cgroupv2)
             {
-                assert(!"Failed to parse cgroup info file contents with sscanf.");
-                goto done;
+                // See https://www.kernel.org/doc/Documentation/cgroup-v2.txt
+                // Look for "0::/some/path"
+                int sscanfRet = sscanf(line,
+                                       "0::%s",
+                                       cgroup_path);
```

Comment:
Don't use `scanf()` with `"%s"` conversion specifier.  It's an unbounded read.  There are ways to limit the number of bytes read from it with precision modifiers; the man page mentions how it works.

Reply:
@omajid, @lpereira  It is not a problem here. The string storage was allocated to be sufficient (size of the whole line we are scanning). The same for the comment in the sscanf below.